### PR TITLE
Hash events collection [Verifiable events part 1]

### PR DESCRIPTION
**Pull request by m4ksio**
Adds EventsList type which captures list of events and hashes them, propagating it to the Chunk.
It's the first part of changes to add verifiable events to the system, kept here for ease of review

### CODE REVIEW

File: model/flow/identifier.go
Comment (ramtinms) [inline model/flow/identifier.go:134]: just a headsup or maybe a comment this Merkle Root computation is very different from the way we do it for exec storage (ledger) this one uses blake2 for hashing.

File: engine/execution/ingestion/engine.go
Comment (ramtinms) [inline engine/execution/ingestion/engine.go:1081]: I don't think it is a good idea to do the operation here, this is not a negligible operation and directly impact TPS slowdown. We probably should do the hashing operation as part of committer and just pass the final root hash to generateChunk.

File: model/flow/identifier.go
Comment (tarakby) [reply model/flow/identifier.go:129]: Btw, can't the key of the tree be the index of the event in the input list? This avoids the extra hashing to compute the ID and allows the tree to remain shallow.

File: engine/execution/ingestion/engine.go
Comment (m4ksio) [reply engine/execution/ingestion/engine.go:1081]: That was planned as a next step, to keep this PR simple

File: model/flow/event.go
Comment (m4ksio) [reply model/flow/event.go:117]: Probably. For now, the plan is to verify all the events emitted, but in the future we would like to have it more granular, but I'm not sure on the details

File: model/flow/event.go
Comment (m4ksio) [reply model/flow/event.go:114]: I'm not aware how we can avoid this copy, but interface types are pointers already AFAIK

File: model/flow/identifier.go
Comment (m4ksio) [reply model/flow/identifier.go:129]: `EventID` is already used in many places, including SDK so I don't want to change semantics.
But now I think we can kill two birds with one stone - have separate tree path which can be used as tree key here, but it will require use to keep the new interface, renaming it a bit

File: engine/execution/ingestion/engine.go
Comment (ramtinms) [reply engine/execution/ingestion/engine.go:1081]: My only concern is if this gets merge into master before the next step and make it to an spork

Comment (m4ksio) [reply engine/execution/ingestion/engine.go:1081]: Sure, I can prepare PR on top of this to before merging this one

File: model/flow/identifier.go
Comment (m4ksio) [reply model/flow/identifier.go]: As discussed, switched to list hashing

File: model/flow/event.go
Comment (m4ksio) [reply model/flow/event.go:50]: Yep - done

File: model/flow/identifier.go
Comment (m4ksio) [reply model/flow/identifier.go]: Switched to hashing list as discussed, so I it doesn't apply anymore

**Review by tarakby — APPROVED** [review]
👌

File: engine/execution/ingestion/engine.go
Comment (m4ksio) [reply engine/execution/ingestion/engine.go]: https://github.com/onflow/flow-go/pull/868

**Review by ramtinms — APPROVED** [review]

**Pull request merged**
